website/source/docs/secrets/pki/index.html.md

```diff
@@ -482,87 +992,95 @@ If you get stuck at any time, simply run `vault path-help pki` or with a subpath
-        `foo.example.com` and `*.example.com`. To allow further
-        levels of subdomains, enable the `allow_subdomains` option.
-        There is no default.
+        **N.B.**: In 0.4+, the meaning of this value has changed, although the
+        name is kept for backwards compatibility.<br /><br />Designates the
+        base domain of the role. This is used with the `allow_base_domain` and
+        `allow_subdomains` options. There is no default.
+      </li>
+      <li>
+        <span class="param">allow_base_domain</span>
+        <span class="param-flags">optional</span>
+        If set, clients can request certificates matching the value of the
+        actual base domain. Defaults to false.
       </li>
       <li>
         <span class="param">allow_token_displayname</span>
         <span class="param-flags">optional</span>
-        If set, clients can request certificates matching
-        the value of Display Name from the requesting token.
-        Remember, this stacks with the other CN options,
-        including `allowed_base_domain`. Defaults to `false`.
+        If set, clients can request certificates matching the value of Display
+        Name from the requesting token. Remember, this stacks with the other CN
+        options, including `allow_subdomains`. Defaults to `false`.
       </li>
       <li>
         <span class="param">allow_subdomains</span>
         <span class="param-flags">optional</span>
-        If set, clients can request certificates with CNs that
-        are subdomains of the CNs allowed by the other role
-        options. _This includes wildcard subdomains._ This is
-        redundant when using the `allow_any_name` option.
-        Defaults to `false`.
+        If set, clients can request certificates with CNs that are subdomains
+        of the CNs allowed by the other role options. _This includes wildcard
+        subdomains._ For example, an `allowed_base_domain` value of
+        `example.com` with this option set to true will allow `foo.example.com`
+        and `bar.example.com` as well as `*.example.com`. This is redundant
+        when using the `allow_any_name` option.  Defaults to `false`.
       </li>
       <li>
         <span class="param">allow_any_name</span>
         <span class="param-flags">optional</span>
-        If set, clients can request any CN. Useful in some
-        circumstances, but make sure you understand whether it
-        is appropriate for your installation before enabling it.
-        Defaults to `false`.
+        If set, clients can request any CN. Useful in some circumstances, but
+        make sure you understand whether it is appropriate for your
+        installation before enabling it.  Defaults to `false`.
+      </li>
+      <li>
+        <span class="param">enforce_hostnames</span>
+        <span class="param-flags">optional</span>
+        If set, only valid host names are allowed for CNs, DNS SANs, and the
```

Comment:
Seems reasonable to default this to `true`. I would think most people are issuing valid CN's generally